### PR TITLE
Bump crio for fixed handling of oom with cgroupv2

### PR DESCRIPTION
**Pull request by mimowo**
It bumps the CRIO used by test-infra to its latest version after the fixes for OOM killer handling, pointing to the new version after https://github.com/cri-o/cri-o/pull/6629.

For more reference check [Add e2e_node test for oom killed container reason](https://github.com/kubernetes/kubernetes/pull/113205) and in particular this recent [comment](https://github.com/kubernetes/kubernetes/pull/113205#issuecomment-1424517837). 

I've tested locally that the OOMKiller test passes.

This is the last analogous commit which bumped CRIO: https://github.com/kubernetes/test-infra/commit/2b30f2e31d90eabcc47f7616e19e3485b09a9f59.

### CODE REVIEW

**Comment by mimowo** [comment]
/assign @SergeyKanzhelev 
FYI @bobbypage @alculquicondor

**Comment by mimowo** [comment]
cc @saschagrunert @haircommander

**Comment by alculquicondor** [comment]
why are we referencing a commit `fa0d058141343ee38b8339ea41f580c361271372` instead of a tagged release? Unless this commit matches a tag, this seems very risky.

**Comment by mimowo** [comment]
> why are we referencing a commit `fa0d058141343ee38b8339ea41f580c361271372` instead of a tagged release? Unless this commit matches a tag, this seems very risky.

The fix hasn't been released yet. From what I could see the previous commit `a8c9ddc773007c61074e224bd740ca3aecd7fecb` also didn't match a released tag. IIUC otherwise we need to wait until the next CRI-O release, but this seems to be in sync with k/k releases so 1.27.0. I guess the general approach is to test against the latest version, but the process of bumping is manual. 

I agree this is risky - if there is a regression in the new version of CRIO, then some tests may fail. I did a sanity test of running the new test locally, but there are too many tests and configurations for me to test. However, even if we bumped to the latest released version there is still a risk that some k/k CRIO tests would fail and we need to revert the commit. This could be even worse because then the maintainers of CRIO would know about the regression late.

Still, I think the tests with CRIO for k/k are not blocking so if some fail after the upgrade then most people won't be affected, except those who want to run the tests. If some tests fail we will need to most likely revert the commit unfortunately, or roll forward if the change is expected.

Pity, there is no way to run all CRIO tests in all configuration prior to merging this PR, which would be very useful also in other cases.

@alculquicondor @haircommander  WDYT, how should we proceed?

**Comment by mimowo** [comment]
> Pity, there is no way to run all CRIO tests in all configuration prior to merging this PR, which would be very useful also in other cases.

IIUC This is not just about CRIO tests, from what I know there is no way to run k/k tests against a branch version of test-infra, so often we need to merge a pull configuration changes before thorough testing, unfortunately, I think this could be improved if there was a way to point in a k/k PR to run tests against a PR in test-infra.

@SergeyKanzhelev has this been considered? Maybe we should open an Issue?

**Comment by haircommander** [comment]
this will need to be rebased on https://github.com/kubernetes/test-infra/pull/28714

> why are we referencing a commit fa0d058141343ee38b8339ea41f580c361271372 instead of a tagged release? Unless this commit matches a tag, this seems very risky. 

yeah this is how we've historically done it. there would be no way to test against the newest cri-o version to get e.g. updated CRI options for new KEPS. CRI-O cuts releases in sync with Kubernetes, so we need a way to test cri-o main branch. It would actually be more ideal to test `main` directly, but we got feedback from sig testing that it should be pinned, so this is our approach

**Comment by alculquicondor** [comment]
I see, thanks for the context.

I'll leave it to sig-testing to approve, but overall the change looks standard procedure.

**Comment by mimowo** [comment]
@haircommander I also bump `jobs/e2e_node/swap/crio_swap1g.ign` which wasn't aligned. Seems like an omission from https://github.com/kubernetes/test-infra/commit/8939aa8308e53b5046cb889f7ba6f7dbc2527e60

**Comment by haircommander** [comment]
/approve

LGTM, @harche PTAL

**Comment by harche** [comment]
@mimowo Thanks for this PR. I was wondering if you were able to test any of these ignition files by manually invoking node e2e locally (which ends up launching the job on gcp using the specified ignition file)

**Comment by mimowo** [comment]
> @mimowo Thanks for this PR. I was wondering if you were able to test any of these ignition files by manually invoking node e2e locally (which ends up launching the job on gcp using the specified ignition file)

@harche  Yes, I tested via the ignition files. I tested it using my GCP project, with a script like this:

```
mkdir -p "/tmp/e2e-node-results"
UUID="$(cat /dev/urandom | tr -dc 'a-z' | fold -w 4 | head -n 1)"
ARTIFACTS="/tmp/e2e-node-results/${UUID}"
mkdir -p "${ARTIFACTS}"
echo "ARTIFACTS= ${ARTIFACTS}"

PROJECT="XYZ" # my project
ZONE="XYZ" # my zone

IMAGE_CONFIG="/workspace/test-infra/jobs/e2e_node/crio/latest/image-config-cgrpv2-k8s-infra-prow-build.yaml"
SSH_KEY="/root/.ssh/google_compute_engine"

docker run \
    --rm \
    --entrypoint=kubetest \
    -it \
    -e ARTIFACTS="${ARTIFACTS}" \
    -e JENKINS_GCE_SSH_PRIVATE_KEY_FILE="/root/.ssh/google_compute_engine" \
    -e JENKINS_GCE_SSH_PUBLIC_KEY_FILE="/root/.ssh/google_compute_engine.pub" \
    -e GCE_SSH_PUBLIC_KEY_FILE="/root/.ssh/google_compute_engine.pub" \
    -e IGNITION_INJECT_GCE_SSH_PUBLIC_KEY_FILE=1 \
    -e KUBE_SSH_USER="core" \
    -w /go/src/k8s.io/kubernetes \
    -v ${HOME}/go/src/k8s.io/kubernetes:/go/src/k8s.io/kubernetes \
    -v ${HOME}/go/src/k8s.io/test-infra:/workspace/test-infra:ro \
    -v ${HOME}/go/src/k8s.io/test-infra:/home/prow/go/src/k8s.io/test-infra:ro \
    -v ~/.config/gcloud:/root/.config/gcloud \
    -v ${HOME}/.ssh/google_compute_engine:/root/.ssh/google_compute_engine:ro \
    -v ${HOME}/.ssh/google_compute_engine.pub:/root/.ssh/google_compute_engine.pub:ro \
    -v ${HOME}/go/src/github.com/containerd/containerd:/home/prow/go/src/github.com/containerd/containerd:ro \
    -v ${HOME}/go/src/github.com/cri-o/cri-o:/home/prow/go/src/github.com/cri-o/cri-o:ro \
    -v ${ARTIFACTS}:${ARTIFACTS} gcr.io/k8s-staging-test-infra/kubekins-e2e:latest-master \
    kubetest \
      --up \
      --down \
      --test \
      --provider=gce \
      --deployment=node \
      --gcp-project="${PROJECT}" \
      --gcp-zone="${ZONE}" \
      --node-args=--image-config-file="${IMAGE_CONFIG}" \
      - '--node-test-args=--container-runtime-endpoint=unix:///var/run/crio/crio.sock --container-runtime-process-name=/usr/local/bin/crio --container-runtime-pid-file= --kubelet-flags="--cgroup-driver=systemd --cgroups-per-qos=true --cgroup-root=/ --runtime-cgroups=/system.slice/crio.service --kubelet-cgroups=/system.slice/kubelet.service" --extra-log="{\"name\": \"crio.log\", \"journalctl\": [\"-u\", \"crio\"]}"' \
      --node-tests=true \
      '--test_args=--nodes=1 --focus=".*OOMKiller.*" --skip="" --timeout=300m --no-color' \
      '--timeout=300m' 2>&1 | tee -i "${ARTIFACTS}/build-log.txt"
```

**Comment by haircommander** [comment]
PTAL @SergeyKanzhelev @mrunalp

**Review by mrunalp — APPROVED** [review]

**Pull request merged**
